spec/models/abuse_report_spec.rb

```diff
@@ -88,38 +86,75 @@
     end
 
-    let(:common_report_variant) { build(:abuse_report, url: work_url_variant) }
-    it "can't be submitted with a variation of the URL" do
-      ArchiveConfig.ABUSE_REPORTS_PER_WORK_MAX.times do
-        create(:abuse_report, url: work_url)
+    shared_examples "enough already" do |url|
+      let(:report) { build(:abuse_report, url: url) }
+      it "can't be submitted" do
+        expect(AbuseReport.count).to eq(ArchiveConfig.ABUSE_REPORTS_PER_WORK_MAX)
+        expect(report.save).to be_falsey
+        expect(report.errors[:base].first).to include("URL has already been reported.")
       end
-      expect(common_report_variant.save).to be_falsey
-      expect(common_report_variant.errors[:base]).not_to be_empty
     end
+
+    shared_examples "alright" do |url|
+      let(:report) { build(:abuse_report, url: url) }
+      it "can be submitted" do
+        expect(AbuseReport.count).to eq(ArchiveConfig.ABUSE_REPORTS_PER_WORK_MAX)
+        expect(report.save).to be_truthy
+        expect(report.errors[:base]).to be_empty
+      end
+    end
+
+    # obviously
+    it_behaves_like "enough already", work_url
+
+    # the same work, with parameters/anchors
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789?smut=yes"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789?smut=yes#timeline"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789#timeline"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/?smut=yes"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/#timeline"
+
+    # the same work, in a collection
+    it_behaves_like "enough already", "http://archiveofourown.org/collections/rarepair/works/789"
+
+    # the same work, subpages
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/bookmarks"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/collections"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/comments"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/kudos"
+
+    # a specific chapter on the work
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/chapters/123"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/chapters/123#major-character-death"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/chapters/123?ending=1"
+    it_behaves_like "enough already", "http://archiveofourown.org/works/789/chapters/123?ending=2#major-character-death"
+
+    # not the same work
+    it_behaves_like "alright", "http://archiveofourown.org/works/9009"
+    it_behaves_like "alright", "http://archiveofourown.org/works/78"
+    it_behaves_like "alright", "http://archiveofourown.org/works/7890"
+    it_behaves_like "alright", "http://archiveofourown.org/comments/show_comments?work_id=789"
```

Comment:
Should this and `it_behaves_like "alright", "http://archiveofourown.org/chapters/123"` maybe be broken out into their own section instead of placed under `# not the same work`? They are technically the same work; they're just variations we didn't think were important to cover at the time.